src/p_user.c

```diff
@@ -316,29 +315,21 @@ void P_DeathThink (player_t* player)
 
   if (player->cmd.buttons & BT_USE)
   {
-    if (demorecording || demoplayback || netgame)
-      player->playerstate = PST_REBORN;
-    else switch(death_use_action)
+    player->playerstate = PST_REBORN;
+  }
+
+  if (activate_death_use_reload)
```

Comment:
Couldn't we just check for `death_use_reload` in the `if (player->cmd.buttons & BT_USE)` case above and also add the demo/netgame checks here? Then we wouldn't need the additional `activate_death_use_reload` variable.

Reply:
But how do we know if the "use" button was pressed without `activate_death_use_reload`? We don't set `BT_USE` to avoid recording it in demo/netgame, so the condition `if (player->cmd.buttons & BT_USE)` will be `false`.

Reply:
Ah, right!